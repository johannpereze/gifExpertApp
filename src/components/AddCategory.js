import { useState } from "react";
export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("Hola Mundo");

  const handleInputChange = ()=>{
      
  }

  return (
    <>
      <input type="text" value={inputValue} onChange={console.log} />
    </>
  );
};
