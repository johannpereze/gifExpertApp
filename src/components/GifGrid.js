// import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types"
require("dotenv").config();

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className={"animate__animated animate__fadeIn"}>{category}</h3>
      {loading && <p>Cargando...</p>}
      <div className={"card-grid"}>
        {images.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};


GifGrid.prototypes ={
  category: PropTypes.string.isRequired
}