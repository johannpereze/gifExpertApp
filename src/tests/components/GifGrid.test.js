import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
import "@testing-library/jest-dom";
jest.mock("../../hooks/useFetchGifs")

describe('Pruebas del componente GifGrid', () => {
    const category = "Evangelion"

    test('DeberÃ­a hacer match con el snapshop', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {

        const gifs = [{
            id: 5346457457,
            title: "titulo",
            url: "https://gfs.com/5346457457.gif",
        },
        {
            id: 53464346346347457,
            title: "titulo555",
            url: "https://gfs.com/534546457457.gif",
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    })


})


