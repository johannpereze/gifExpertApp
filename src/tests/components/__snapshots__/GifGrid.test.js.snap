// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGrid Debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Evangelion
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={5346457457}
      key="5346457457"
      title="titulo"
      url="https://gfs.com/5346457457.gif"
    />
    <GifGridItem
      id={53464346346347460}
      key="53464346346347460"
      title="titulo555"
      url="https://gfs.com/534546457457.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas del componente GifGrid Deber√≠a hacer match con el snapshop 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Evangelion
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
